src/NHapi.Base/Model/AbstractGroup.cs

```diff
@@ -269,20 +256,37 @@ public virtual string AddNonstandardSegment(string name)
                 throw new HL7Exception("Need message version to add segment by name; message.getVersion() returns null");
             }
 
-            var c = myFactory.GetSegmentClass(name, version);
-            if (c == null)
+            var c = myFactory.GetSegmentClass(name, version) ?? typeof(GenericSegment);
+
+            var index = Names.Length;
+
+            TryToInstantiateStructure(c, name); // may throw exception
+
+            return Insert(c, false, true, false, index, name);

```

Comment:
I think it would be nice to add the param names so its more obvious what `true` and `false1 represent.
example...
```csharp
return Insert(
    segment: c,
    whateverThisIsCalled: false
...
...
etc
```

Reply:
I already see this (must be resharper):
![image](https://user-images.githubusercontent.com/1087376/114867565-e1bc7880-9dec-11eb-8b30-dcb3e59dcfad.png)
I think I'm going to leave this one for now.